packages/js/data/src/crud/utils.ts

```diff
@@ -168,3 +215,46 @@ export const cleanQuery = (
 
 	return cleaned;
 };
+
+/**
+ * Get the identifier for a request provided its arguments.
+ *
+ * @param  name Name of action or selector.
+ * @param  args Arguments for the request.
+ * @return Key to identify the request.
+ */
+export const getRequestIdentifier = ( name: string, ...args: unknown[] ) => {
+	const suffix = JSON.stringify(
+		args.map( ( arg ) => {
+			if ( typeof arg === 'object' && arg !== null ) {
```

Comment:
Important to note here that the `getResourceName` util in the data package does not accurately work for checking request status.  It requires us to name all arguments and can only accept a single argument.

This util only cares about the order of the arguments provided, but does sort object properties to help mitigate duplicate calls.